subsys/bluetooth/host/hci_core.c

```diff
@@ -329,6 +329,15 @@ int bt_hci_cmd_send_sync(uint16_t opcode, struct net_buf *buf,
 
 	net_buf_put(&bt_dev.cmd_tx_queue, net_buf_ref(buf));
 
+	/* Wait for a response from the Bluetooth controller.
+	 * The controller may fail to respond if:
+	 *  - It was never programmed or connected.
+	 *  - It has crashed.
+	 *
+	 * See Core_v5.4, Vol 4, Part E, Section 5.4.1 and Section 7 to map
+	 * the opcode to the HCI command documentation.
+	 * Example: 0x0c03 represents HCI_Reset command.
```

Comment:
It is also possible to grep `hci_types.h` for the opcodes.

Reply:
Added a reference to that file here as well. Better to be more clear.

Reply:
This is good documentation improvement Since sometimes HCI_RESET throws hardfault. The hardfault causes confusion in debugging direction. This clear documenation will clearly help developers wandering here.

Reply:
Another common reason this can happen is if there's a bug somewhere and the Bluetooth-related threads (system wq included) are deadlocked. Might also be worth mentioning.

Reply:
I have certainly deadlocked the Bluetooth Host many times from doing silly things in applications.
Application level Bluetooth control in the system workqueue can be tricky to get right.

Reply:
> may make people believe the Bluetooth Host is likely to deadlock from time to time

Yes, but as Jordan says, it's oft the fault of the application. This is one of the cons of callback-based API. It's hard to really assign blame here, as the list of no-nos in host callbacks is documented nowhere.

For example I recently debugged in nordic an issue where the app would get the `disconnected` callback and then call a host API to get the current device RPA. This triggered an RPA refresh, trying to send an HCI command to the LL. The problem is that we were out of cmd buffers, and the free would happen later on the same workqueue. It was not clear from the documentation of the address-get API that it is not OK to call it in a host callback.

![image](https://github.com/zephyrproject-rtos/zephyr/assets/68428270/79c008b1-eae4-43b8-b2d8-d75f0054967c)

Reply:
Discussed offline. Will handle this separately as we need some documentation that can be referred to.